pancloud/credentials.py

```diff
@@ -271,19 +267,9 @@ def get_credentials(self):
             class: Read-only credentials.
 
         """
-        if self.cache_token:
-            access_token = self.access_token_ or \
-                           self._resolve_credential('access_token')
-        else:
-            access_token = self.access_token_
-        client_id = self.client_id_ or self._resolve_credential(
-            'client_id')
-        client_secret = self.client_secret_ or self._resolve_credential(
-            'client_secret')
-        refresh_token = self.refresh_token_ or self._resolve_credential(
-            'refresh_token')
         return ReadOnlyCredentials(
-            access_token, client_id, client_secret, refresh_token
+            self.access_token, self.client_id, self.client_secret,
```

Comment:
shouldn't the self names end with _ ?

Reply:
also see line 300.  may be missing something.

Reply:
1. The idea is that we're accessing the property methods which will either return the value of the instance variable, i.e. the variables that end with _, or the value returned by `_resolve_credential()` method.

2. The idea behind line 300 is that we only refresh if the passed in `access_token` is the same as the one returned by the property method or if it's `None`. We compare it to the value returned by the property method to avoid refreshing again, i.e. another thread already completed a `refresh()` and somehow the existing thread was able to acquire the lock.